### PR TITLE
[FIX JENKINS-45857] Support to set the Binding type (Redirection/Post)

### PR DESCRIPTION
**Pull request by kuisathaverat**
[JENKINS-45857](https://issues.jenkins-ci.org/browse/JENKINS-45857)

It adds a new setting to select the binding method if it's not selected or it's an old configuration the default value is HTTP-Redirect.

@reviewbybees

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -28,13 +28,40 @@ Report new issue on https://issues.jenkins-ci.org on component **saml-plugin**.
...
 
     * org.jenkinsci.plugins.saml - FINEST
     * org.pac4j - FINE
-    
-Changelog
--------------------
-[Changelog](CHANGELOG.md)
+
+
+**Identity provider has no single sign on service available for the selected...**
```
Comment (amuniz) [inline README.md]: ... profile (?)

File: src/main/java/org/jenkinsci/plugins/saml/SamlPluginConfig.java [hunk 2]
```diff
@@ -36,13 +36,14 @@
...
     private final String logoutUrl;
+    private final String binding;
 
     private SamlEncryptionData encryptionData;
     private SamlAdvancedConfiguration advancedConfiguration;
 
     public SamlPluginConfig(String displayNameAttributeName, String groupsAttributeName,
                             int maximumAuthenticationLifetime, String emailAttributeName, String idpMetadata,
-                            String usernameCaseConversion, String usernameAttributeName, String logoutUrl,
+                            String usernameCaseConversion, String usernameAttributeName, String logoutUrl, String binding,
```
Comment (amuniz) [inline src/main/java/org/jenkinsci/plugins/saml/SamlPluginConfig.java]: Is this change backward compatible on config files?

File: src/test/java/org/jenkinsci/plugins/saml/SamlSecurityRealmTest.java [hunk 3]
```diff
@@ -78,6 +90,20 @@ public void testReadSimpleConfiguration() {
         assertEquals("urn:mace:dir:attribute-def:mail", samlSecurityRealm.getEmailAttributeName());
         assertEquals("urn:mace:dir:attribute-def:uid", samlSecurityRealm.getUsernameAttributeName());
         assertEquals(true, samlSecurityRealm.getIdpMetadata().startsWith("<?xml version"));
+        assertEquals(SAML2_REDIRECT_BINDING_URI, samlSecurityRealm.getBinding());
+    }
+
+    @LocalData
+    @Test
+    public void testReadSimpleConfigurationHTTPPost() {
```
Comment (amuniz) [inline src/test/java/org/jenkinsci/plugins/saml/SamlSecurityRealmTest.java]: Is this test checking backward compatibility? IOW does the `LocalData` contain _old_ config files?

File: README.md (hunk 1 above)
Comment (kuisathaverat) [reply README.md]: it is right IdP (Identity Provider)

File: src/main/java/org/jenkinsci/plugins/saml/SamlPluginConfig.java (hunk 2 above)
Comment (kuisathaverat) [reply src/main/java/org/jenkinsci/plugins/saml/SamlPluginConfig.java]: yes, it is tested manually and in the tests

File: src/test/java/org/jenkinsci/plugins/saml/SamlSecurityRealmTest.java (hunk 3 above)
Comment (kuisathaverat) [reply src/test/java/org/jenkinsci/plugins/saml/SamlSecurityRealmTest.java]: testReadSimpleConfiguration uses an old config file without the field

File: src/main/java/org/jenkinsci/plugins/saml/SamlSecurityRealm.java [hunk 4]
```diff
@@ -178,21 +184,16 @@ public Object readResolve() {
...
-            String groupsAttributeName,
-            Integer maximumAuthenticationLifetime,
-            String usernameAttributeName,
-            String emailAttributeName,
-            String logoutUrl,
-            SamlAdvancedConfiguration advancedConfiguration,
-            SamlEncryptionData encryptionData) throws IOException {
-        this(idpMetadata, displayNameAttributeName, groupsAttributeName, maximumAuthenticationLifetime,
-                usernameAttributeName, emailAttributeName, logoutUrl, advancedConfiguration, encryptionData, "none");
+        if(StringUtils.isEmpty(getBinding())){
```
Comment (kuisathaverat) [inline src/main/java/org/jenkinsci/plugins/saml/SamlSecurityRealm.java]: here is the readResolve() fix to be compatible with old configurations.

**Pull request merged**